x/bank/v2/module.go

```diff
@@ -0,0 +1,109 @@
+package bankv2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc"
+
+	appmodulev2 "cosmossdk.io/core/appmodule/v2"
+	"cosmossdk.io/core/registry"
+	"cosmossdk.io/x/bank/v2/keeper"
+	"cosmossdk.io/x/bank/v2/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+// ConsensusVersion defines the current x/bank/v2 module consensus version.
+const ConsensusVersion = 1
+
+var (
+	_ module.HasGRPCGateway = AppModule{}
+
+	_ appmodulev2.AppModule             = AppModule{}
+	_ appmodulev2.HasGenesis            = AppModule{}
+	_ appmodulev2.HasRegisterInterfaces = AppModule{}
+)
+
+// AppModule implements an application module for the bank module.
+type AppModule struct {
+	cdc    codec.Codec
+	keeper *keeper.Keeper
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(cdc codec.Codec, keeper *keeper.Keeper) AppModule {
+	return AppModule{
+		cdc:    cdc,
+		keeper: keeper,
+	}
+}
+
+// IsAppModule implements the appmodule.AppModule interface.
+func (am AppModule) IsAppModule() {}
+
+// Name returns the bank module's name.
+// Deprecated: kept for legacy reasons.
+func (AppModule) Name() string { return types.ModuleName }
+
+// ConsensusVersion implements HasConsensusVersion
+func (AppModule) ConsensusVersion() uint64 { return ConsensusVersion }
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the bank module.
+func (AppModule) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *gwruntime.ServeMux) {
+	if err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)); err != nil {
+		panic(err)
+	}
+}
```

Comment:
i think in v2 of modules we should decide if we are keeping networking as part of modules or not. This is post 52 release so we should look at reducing the complexity of writing a module.

Reply:
Removed 👍🏾